### PR TITLE
No invalid constructor super/4624

### LINKED ISSUE #4624: 🐛 lint/rules/noInvalidConstructorSuper breaks in typescript when using generics and interfaces
### Environment information

```block
default environment
```

### What happened?

Repro in [playground :)](
https://docs.rome.tools/playground/?code=YwBsAGEAcwBzACAATABvAG4AZwBDAGwAYQBzAHMATgBhAG0AZQA8AEEALABCAD4AIAB7ACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAYQA6ACAAbgB1AG0AYgBlAHIAKQB7AGEAOwB9ACAAfQAKAAoAaQBuAHQAZQByAGYAYQBjAGUAIABJACAAewAKACAAIABhADoAIABuAHUAbQBiAGUAcgA7AAoAfQAKAAoAdAB5AHAAZQAgAEwAbwBuAGcARwBlAG4AZQByAGkAYwBQAGEAcgBhAG0AZQB0AGUAcgBBACAAPQAgACcATABvAG4AZwBHAGUAbgBlAHIAaQBjAFAAYQByAGEAbQBlAHQAZQByAEEAJwAKAHQAeQBwAGUAIABMAG8AbgBnAEcAZQBuAGUAcgBpAGMAUABhAHIAYQBtAGUAdABlAHIAQgAgAD0AIAAnAEwAbwBuAGcARwBlAG4AZQByAGkAYwBQAGEAcgBhAG0AZQB0AGUAcgBCACcACgAKAGMAbABhAHMAcwAgAEYAYQBpAGwACgAJAGUAeAB0AGUAbgBkAHMAIABMAG8AbgBnAEMAbABhAHMAcwBOAGEAbQBlADwATABvAG4AZwBHAGUAbgBlAHIAaQBjAFAAYQByAGEAbQBlAHQAZQByAEEALAAgAEwAbwBuAGcARwBlAG4AZQByAGkAYwBQAGEAcgBhAG0AZQB0AGUAcgBCAD4ACgAJAGkAbQBwAGwAZQBtAGUAbgB0AHMAIABJAAoAewAKAAkALwAvACAAdwBoAGUAbgAgAHQAaABpAHMAIABpAHMAIABvAG4AIABhACAAcwBpAG4AZwBsAGUAIABsAGkAbgBlACwAIABzAHUAcABlAHIAIABoAGEAcwAgAG4AbwAgAGUAcgByAG8AcgAgACgAZQB4AHAAZQBjAHQAZQBkACkACgAJAC8ALwAgAGIAdQB0ACAAdwBoAGUAbgAgAHQAaABlACAAZQB4AHAAcgBlAHMAcwBpAG8AbgAgAGEAZgB0AGUAcgAgAGUAeAB0AGUAbgBkAHMAIABpAHMAIABsAG8AbgBnACAAZQBuAG8AdQBnAGgALAAgAGkAdAAgAGcAZQB0AHMAIABsAGkAbgBlACAAdwByAGEAcABwAGUAZAAKAAkALwAvACAAYQBuAGQAIAB0AGgAZQAgAGwAaQBuAHQAZQByACAAZgBhAGkAbABzAAoACQBjAG8AbgBzAHQAcgB1AGMAdABvAHIAKAApACAAewAKAAkACQBzAHUAcABlAHIAKAAxACkAOwAKAAkAfQAKAH0A)

### Expected result

It should not throw an error

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

Fixes #4624.
This PR should prevent future related bugs.

## Test Plan

Regression tests included.

### CODE REVIEW

**Review by ematipico — APPROVED** [review]

**Comment by Aetherall** [comment]
Wow most reactive open source project medal for you @Conaclos  :1st_place_medal: :star_struck:

File: crates/rome_js_analyze/tests/specs/correctness/noInvalidConstructorSuper/valid.ts [hunk 1]
```diff
@@ -41,3 +41,38 @@ export class A extends mod.B {
         super();
     }
 }
+
+// Regression test for https://github.com/rome/tools/issues/4624
+class ExtendGeneric extends A<number> implements I {
```
Comment (Aetherall) [inline crates/rome_js_analyze/tests/specs/correctness/noInvalidConstructorSuper/valid.ts:46]: You might want add a test with long names as my bug only triggered for me when the definition went multiline :)

image

Comment (Aetherall) [reply crates/rome_js_analyze/tests/specs/correctness/noInvalidConstructorSuper/valid.ts:46]: Seems to be fixed on your branch anyway !
image

Comment (Conaclos) [reply crates/rome_js_analyze/tests/specs/correctness/noInvalidConstructorSuper/valid.ts]: Thanks for the suggestion! I expanded the code on multiple lines :)

**Pull request merged**